plugins/spi/SPIOutput.cpp

```diff
@@ -486,6 +500,100 @@ uint8_t SPIOutput::P9813CreateFlag(uint8_t red, uint8_t green, uint8_t blue) {
   return ~flag;
 }
 
+
+void SPIOutput::IndividualAPA102Control(const DmxBuffer &buffer) {
+  // some detailed information on the protocol:
+  // https://cpldcpu.wordpress.com/2014/11/30/understanding-the-apa102-superled/
+  // Data-Struct
+  // StartFrame: 4Byte = 32Bits zeros
+  // LEDFrame: 1Byte FF ; 3Byte color info (Blue, Green, Red)
+  // EndFrame: (n/2)bits; n = pixel_count
+  
+  // i don't know how this works.. but it works ;-)
+  const uint8_t latch_bytes = 3 * APA102_SPI_BYTES_PER_PIXEL;
+  const unsigned int first_slot = m_start_address - 1;  // 0 offset
+  
+  // only do something if minimum 1pixel can be updated..
+  if (buffer.Size() - first_slot < APA102_SLOTS_PER_PIXEL) {
+    // not even 3 bytes of data, don't bother updating
+    return;
+  }
+  
+  // We always check out the entire string length, even if we only have data
+  // for part of it
+  const unsigned int output_length = m_pixel_count * APA102_SPI_BYTES_PER_PIXEL;
```

Comment:
Shouldn't this be (m_pixel_count + 1) since you need an extra 4 bytes at the start?